portal-api/api-tests/v1/companies/companies.api-test.js

```diff
@@ -0,0 +1,99 @@
+const { MOCK_COMPANY_REGISTRATION_NUMBERS } = require('@ukef/dtfs2-common');
+const app = require('../../../src/createApp');
+const { MAKER, READ_ONLY, ADMIN } = require('../../../src/v1/roles/roles');
+const testUserCache = require('../../api-test-users');
+const { withClientAuthenticationTests } = require('../../common-tests/client-authentication-tests');
+const { withRoleAuthorisationTests } = require('../../common-tests/role-authorisation-tests');
+const { POSTCODE } = require('../../fixtures/postcode');
+const { companies } = require('../../../src/external-api/api');
+const { as, get } = require('../../api')(app);
+
+jest.unmock('../../../src/external-api/api');
+
+describe.each([{ baseUrl: '/v1' }, { baseUrl: '/v1/gef' }])('GET $baseUrl/companies/:registrationNumber', ({ baseUrl }) => {
+  let testUsers;
+  let aMaker;
+
+  beforeAll(async () => {
+    testUsers = await testUserCache.initialise(app);
+    aMaker = testUsers().withRole(MAKER).one();
+  });
+
+  it('returns a 200 response with the company when it is found', async () => {
+    const { status, body } = await as(aMaker).get(`${baseUrl}/companies/${MOCK_COMPANY_REGISTRATION_NUMBERS.VALID}`);
+    expect(status).toEqual(200);
+    expect(body.companiesHouseRegistrationNumber).toEqual(expect.any(String));
+    expect(body.companyName).toEqual(expect.any(String));
+    expect(body.registeredAddress).toEqual({
+      addressLine1: expect.any(String),
+      addressLine2: expect.toBeStringOrUndefined(),
+      addressLine3: expect.toBeStringOrUndefined(),
+      locality: expect.any(String),
+      postalCode: expect.any(String),
+      country: expect.any(String),
+    });
+    expect(body.industries[0].code).toEqual(expect.any(String));
+    expect(body.industries[0].name).toEqual(expect.any(String));
+    expect(body.industries[0].class).toEqual({
+      code: expect.any(String),
+      name: expect.any(String),
+    });
+  });
+
+  it('returns a 400 response if an invalid company registration number is provided', async () => {
+    const { status } = await as(aMaker).get(`${baseUrl}/companies/${MOCK_COMPANY_REGISTRATION_NUMBERS.INVALID_TOO_SHORT}`);
+    expect(status).toEqual(400);
+  });
+
+  it('returns a 404 response when the company is not found', async () => {
+    const { status } = await as(aMaker).get(`${baseUrl}/companies/${MOCK_COMPANY_REGISTRATION_NUMBERS.VALID_NONEXISTENT}`);
+    expect(status).toEqual(404);
+  });
+
+  it('returns a 422 response when the company is an overseas company', async () => {
```

Comment:
Can we add `429` and `500` please?

Reply:
As mentioned in last Dev Friday, at the moment the API tests for Portal API don't mock External API, meaning that all API calls are actually made. I will raise a ticket for this as it needs to be changed across Portal API. So at the moment, we have no way of simulating a `429` or a `500` because the API calls are for real. Once we've changed these tests to mock External API, we can cover more test cases such as these (I'll add this to the ticket I raise).

Reply:
Understood and thank you.